src/main/java/fr/zcraft/zlib/components/gui/ArrayPromptGui.java

```diff
@@ -0,0 +1,93 @@
+package fr.zcraft.zlib.components.gui;
+
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+
+import fr.zcraft.zlib.components.gui.ExplorerGui;
+import fr.zcraft.zlib.components.gui.Gui;
+import fr.zcraft.zlib.tools.Callback;
+
+/* Built from recommendations in PR https://github.com/zDevelopers/zLib/pull/14 */
+public class ArrayPromptGui<T> extends ExplorerGui<T> {
+
+	private Callback<T> cb;
+	private String purpose;
+	private T[] datas;
+	private ItemRenderer<T> renderer;
+	private boolean closeOnChoice;
+
+	/**
+	 * @param callback
+	 *            Callback called when the player made a choice
+	 * @param player
+	 *            The player making the choice
+	 * @param purpose
+	 *            The purpose (gui's title)
+	 * @param datas
+	 *            An array of datas to display
+	 * @param renderer
+	 *            Interface for building gui's ItemStack from array values
+	 * @param closeOnChoice
+	 *            Close the interface when the player has choosen if true
+	 */
+	public <A> ArrayPromptGui(Callback<T> callback, Player player, String purpose, T[] datas, ItemRenderer<T> renderer,
```

Comment:
You can also add both a callback in the constructor and as a method to override, the method defaulting to executing the stored callback if it's not `null`. *The Return of the Lambda*.